modules/programs/vscode.nix

```diff
@@ -34,6 +34,11 @@ let
   # TODO: On Darwin where are the extensions?
   extensionPath = ".${extensionDir}/extensions";
 
+  mergedUserSettings = cfg.userSettings
+    // optionalAttrs (cfg.disableUpdateCheck) { "update.mode" = "none"; }
+    // optionalAttrs (cfg.disableExtensionUpdateCheck) {
+      "extensions.autoCheckUpdates" = false;
```

Comment:
IMO it would be better to set these options in the config section:

```nix
{
  config.programs.vscode.userSettings = {
    "update.mode" = mkIf cfg.disableUpdateCheck "none";
    "extensions.autoCheckUpdates" = mkIf cfg.disableExtensionUpdateCheck false;
  };
}
```

Reply:
I think this might lead to undesired results when some userSettings are passed in with an existing config and this module is expected to update only the relevant settings.